src/qaqc/download_and_flag_fxns/add_frozen_flag.R

```diff
@@ -0,0 +1,20 @@
+add_frozen_flag <- function(df){
+
+  # create a df of temperature for each site
+  temperature <- df %>%
+    data.table::data.table() %>%
+    dplyr::select(DT_round, DT_join, parameter, mean) %>%
+    dplyr::filter(parameter == "Temperature") %>%
+    dplyr::select(DT_join, Temperature = mean)
+
+    # add "temperature" column to df:
+    temperature_checked <- df %>%
+      dplyr::left_join(., temperature, by = "DT_join") %>%
+      # If water temperature is freezing, flag all parameters
+      add_flag(., Temperature <= 0 & !grepl("frozen", flag), "frozen") %>%
```

Comment:
question: I think that we don't have to add the `& !grepl("frozen", flag)` to the flags, that should be handled in the add_flag() function. Have you noticed that it is not working properly?

Reply:
I'm not sure... this could be overkill. This IS happening across all flags currently (where we make sure the flag doesn't already exist in the observation). We should double-check this.